src/cluster_legacy.c

```diff
@@ -3578,6 +3742,25 @@ static void clusterBuildMessageHdr(clusterMsg *hdr, int type, size_t msglen) {
     hdr->totlen = htonl(msglen);
 }
 
+static void clusterBuildPubsubMessageHdr(clusterMsgLight *hdr, int type, size_t msglen) {
```

Comment:
This is unrelated to pubsub, right? Maybe we can name it as `clusterBuildLightMessageHdr`. Also, we could reuse most of these code across the two methods. Could we have a base method to avoid redundancy and invoked from `clusterBuildMessageHdr` and `clusterBuildLightMessageHdr`?